strawberryfields/compilers/gaussian_unitary.py

```diff
@@ -18,18 +18,70 @@
 from strawberryfields import ops
 from strawberryfields.parameters import par_evaluate
 from thewalrus.symplectic import (
-    expand_vector,
     expand,
     rotation,
     squeezing,
     two_mode_squeezing,
     interferometer,
     beam_splitter,
 )
-
 from .compiler import Compiler
 
 
+def _apply_symp_one_mode_gate(S_G, S, r, i):
+    """In-place applies a one mode gate G to the symplectic operation, S, in mode i
+    Args:
+        S_G (array): 2x2 matrix for one mode symplectic operation
+        S (array): 2Mx2M Symplectic operation
+        r (array): M length means vector
+        i (int): index of one mode gate
+
+    Returns:
+        S (array): updated symplectic operation
+        r (array): updated means vector
+    """
+    M = S.shape[0] // 2
+    (S[i], S[i + M]) = (
+        S_G[0, 0] * S[i] + S_G[0, 1] * S[i + M],
+        S_G[1, 0] * S[i] + S_G[1, 1] * S[i + M],
+    )
+    (r[i], r[i + M]) = (
+        S_G[0, 0] * r[i] + S_G[0, 1] * r[i + M],
+        S_G[1, 0] * r[i] + S_G[1, 1] * r[i + M],
+    )
+    return S, r
+
+
+def _apply_symp_two_mode_gate(S_G, S, r, i, j):
+    """In-place applies a two mode gate G to the symplectic operation, S, in modes i and j
+
+    Args:
+        S_G (array): 2x2 matrix for one mode symplectic operation
+        S (array): 2Mx2M Symplectic operation
+        r (array): M length means vector
+        i (int): index of one mode gate
+        j (int): index of second mode of gate
+
+    Returns:
+        S (array): updated symplectic operation
+        r (array): updated means vector
```

Comment:
Suggest making the same modifications as my comment above!

Reply:
done!